sources/LibProtection.Injections/FormatProvider.cs

```diff
@@ -1,47 +1,22 @@
-﻿using LibProtection.Injections.Caching;
-using System;
+﻿using System;
 using System.Collections.Generic;
 using System.Diagnostics;
 using System.Text;
 
 namespace LibProtection.Injections
 {
-    internal class FormatProvider : IFormatProvider
+    internal class FormatProvider<T> : IFormatProvider where T: LanguageProvider
     {
-        private readonly Formatter _formatter;
+        // ReSharper disable once StaticMemberInGenericType
+        private static volatile ICache<FormatCacheItem, Option<string>> _cache
+            = new RandomizedLRUCache<FormatCacheItem, Option<string>>(1024);
+
+        private readonly Formatter<T> _formatter;
         private readonly List<Fragment> _formattedFragments = new List<Fragment>();
 
         private FormatProvider(char complementaryChar)
         {
-            _formatter = new Formatter(complementaryChar);
-        }
-
-        private static class GenericHolder<T> where T : LanguageProvider
-        {
-            // ReSharper disable once StaticMemberInGenericType
-            public static volatile RandomizedLRUCache<CacheFormatItem, (bool Success, string ResultValue)> Instance
-                = new RandomizedLRUCache<CacheFormatItem, (bool Success, string ResultValue)>(1024);
-
-            // ReSharper disable once StaticMemberInGenericType
-            public static volatile ICustomCache CustomCache;
```

Comment:
removed cache customization? forever? no more custom caches support? :(

Reply:
Similar to the previous :)